### PR TITLE
Remove pull request code

### PR DESCRIPTION
**Pull request by DmitrySharabin**
Partial fix for #367

I also added two hooks (their names are just a hint for me where exactly I put them), so we need to find suitable names for them. Which is hard, you know. 😜
All existing tests pass. But as always, I'm afraid I could miss something. 😅

### CODE REVIEW

File: src/backend.github.js [hunk 1]
```diff
@@ -323,15 +184,12 @@ var _ = Mavo.Backend.register($.Class({
...
-										head: `${this.user.username}:${this.branch}`,
-										base: repoInfo.parent.default_branch
-									}).then(prs => {
-										this.pullRequest(prs[0]);
-									});
-									return repoInfo;
+								if (repoInfo.fork) { // if current repo is a fork, we can display PR dialog
+									const env = { context: this, repoInfo };
+								
+									Mavo.hooks.run("gh-login-no-source-pr", env);
```
Comment (DmitrySharabin) [inline src/backend.github.js:190]: And here comes the second one

File: src/backend.github.js [hunk 2]
```diff
@@ -147,151 +147,12 @@ var _ = Mavo.Backend.register($.Class({
...
-					username = this.repoInfo.parent.owner.login;
-					repo = this.repoInfo.parent.name;
-					base = this.repoInfo.parent.default_branch;
-				}
-				else { // Storage points to another user's repo
-					username = this.username;
-					repo = this.repo;
-					base = this.branch;
-				}
+				Mavo.hooks.run("gh-put-pr", env);
```
Comment (DmitrySharabin) [inline src/backend.github.js:152]: Here is the first hook

File: src/backend.github.js (hunk 1 above)
Comment (DmitrySharabin) [inline src/backend.github.js:190]: And here comes the second one

**Comment by LeaVerou** [comment]
Hi Dmitry,

Thanl you for working on this! This is a good start.

A few first observations:
- The hooks need to be related to the code they're in, not the plugin. We are not adding hooks *specifically* to make the PR plugin possible, so they cannot be named after pull requests. 
- Similarly, the remaining code cannot know about pull requests, otherwise there is no separation. 
- The UX also needs to still make sense without the plugin. People still need to be able to save forks, and be notified that they're in a fork. From a quick review of the code I can't tell for sure what happens when the plugin is not loaded. Have you tested the interaction?
- It's hard to review this without reviewing the plugin code as well. If we are going to be iterating in this PR, perhaps you could add the plugin code at the end of the file, so that they're both in one place, then once it's good to go, you move it to an actual plugin. What do you think? Alternatively, we could have two open pull requests, that works too.

**Comment by DmitrySharabin** [comment]
Hi Lea,

Thanks for your comments. I went through them and worked with the code a bit more. I renamed the hooks and changed the place for one of them. So now there is not a single mention of PRs. I tested both: the backend and the plugin. The backend is fully functional without the plugin, so people still able to save forks and be notified that they're in a fork.

I sent a PR with the plugin's draft to `mavoweb/plugins`. Please have a look.

**Comment by DmitrySharabin** [comment]
Here is another thing I was thinking about. Should we add mirror hooks? I mean, if we have a `gh-after-commit` hook, should we also add `gh-before-commit` (or should we call them `gh-commit-end` and `gh-commit-start` respectively)? I know it depends on whether we want devs to change some core behavior at some point and interfere into, for example, the data flow, but still.

**Comment by LeaVerou** [comment]
Looks good to me, merged.
I don't think we should add hooks that we don't have use cases for.

**Comment by LeaVerou** [comment]
We should also add this plugin on existing Mavos on our sites where we want to accept PRs, e.g. plugins.
Also, we should edit the docs about storage, if they say that PRs are enabled automatically.

**Comment by DmitrySharabin** [comment]
Sure. I'll look carefully at where to make changes.

**Pull request merged**